polaris/competition/_base.py

```diff
@@ -0,0 +1,229 @@
+        n_test_sets: The number of test sets defined in the test split.
+        n_test_datapoints: The number of datapoints in each test set.
+        n_classes: The number of classes within target columns that define a classification task.
+    For additional meta-data attributes, see the [`DatasetV2`][polaris.experimental._dataset_v2.DatasetV2] class.
+    """
+
+    _artifact_type = "competition"
+
+    target_cols: ColumnsType
+    input_cols: ColumnsType
+    split: SplitType
+    metrics: Union[str, Metric, list[str | Metric]]
+    main_metric: str | Metric
+    readme: str
+    target_types: dict[str, Union[TargetType, str]] = Field(default_factory=dict, validate_default=True)
+    start_time: datetime
+    end_time: datetime
+    n_test_sets: int
+    n_test_datapoints: dict[str, int]
+    n_classes: dict[str, int]
+
+    @field_validator("metrics")
+    def _validate_metrics(cls, v) -> list[Metric]:
+        """
+        Verifies all specified metrics are either a Metric object or a valid metric name.
+
+        If there are multiple test sets, it is assumed the same metrics are used across test sets.
+        """
+        if not isinstance(v, list):
+            v = [v]
+
+        v = [m if isinstance(m, Metric) else Metric[m] for m in v]
+
+        return v
+
+    @field_validator("main_metric")
+    def _validate_main_metric(cls, v) -> Metric:
+        """Converts the main metric to a Metric object if it is a string."""
+        # lu: v can be None.
+        if v and not isinstance(v, Metric):
+            v = Metric[v]
+        return v
+
+    @field_serializer("metrics", "main_metric")
+    def _serialize_metrics(self, v) -> list[str] | str:
+        """Return the string identifier so we can serialize the object"""
+        if isinstance(v, Metric):
+            return v.name
+        return [m.name for m in v]
+
+    @field_serializer("target_types")
+    def _serialize_target_types(self, v) -> dict[str, str]:
+        """Convert from enum to string to make sure it's serializable"""
+        return {k: v for k, v in self.target_types.items()}
+
+    @field_serializer("start_time", "end_time")
+    def _serialize_times(self, v: datetime) -> str:
+        """Convert from datetime to string to make sure it's serializable"""
+        return v.isoformat()
+
+    @computed_field
+    @property
+    def task_type(self) -> str:
+        """The high-level task type of the benchmark."""
+        v = TaskType.MULTI_TASK if len(self.target_cols) > 1 else TaskType.SINGLE_TASK
+        return v.value
+
+    @computed_field
+    @property
+    def test_set_labels(self) -> list[str]:
+        """The labels of the test sets."""
+        return sorted(list(self.n_test_datapoints.keys()))
+
+    @computed_field
+    @property
+    def test_set_sizes(self) -> dict[str, int]:
+        """The sizes of the test sets."""
+        return {k: v for k, v in self.n_test_datapoints.items()}
```

Comment:
**Suggestion:**

```suggestion
        return {k: len(v) for k, v in self.split[1].items()}
```

Reply:
This is a remnant of the first version where there was no split on the client side. These achieve the same result though, don't they?

Reply:
Yes, but logically it makes more sense to consider the split as the ground truth, since that actually affects the benchmarking logic.